python_modules/dagster/dagster/_core/definitions/op_invocation.py

```diff
@@ -457,36 +478,19 @@ def _type_check_output(
     from ..execution.plan.execute_step import do_type_check
 
     op_label = context.describe_op()
-
-    if isinstance(output, (Output, DynamicOutput)):
-        dagster_type = output_def.dagster_type
-        type_check = do_type_check(context.for_type(dagster_type), dagster_type, output.value)
-        if not type_check.success:
-            raise DagsterTypeCheckDidNotPass(
-                description=(
-                    f'Type check failed for {op_label} output "{output.output_name}" - '
-                    f'expected type "{dagster_type.display_name}". '
-                    f"Description: {type_check.description}"
-                ),
-                metadata=type_check.metadata,
-                dagster_type=dagster_type,
-            )
-
-        context.observe_output(
-            output_def.name, output.mapping_key if isinstance(output, DynamicOutput) else None
+    dagster_type = output_def.dagster_type
+    type_check = do_type_check(context.for_type(dagster_type), dagster_type, output.value)
+    if not type_check.success:
+        raise DagsterTypeCheckDidNotPass(
+            description=(
+                f'Type check failed for {op_label} output "{output.output_name}" - '
+                f'expected type "{dagster_type.display_name}". '
+                f"Description: {type_check.description}"
+            ),
+            metadata=type_check.metadata,
+            dagster_type=dagster_type,
         )
-        return output
-    else:
-        dagster_type = output_def.dagster_type
-        type_check = do_type_check(context.for_type(dagster_type), dagster_type, output)
-        if not type_check.success:
-            raise DagsterTypeCheckDidNotPass(
-                description=(
-                    f'Type check failed for {op_label} output "{output_def.name}" - '
-                    f'expected type "{dagster_type.display_name}". '
-                    f"Description: {type_check.description}"
-                ),
-                metadata=type_check.metadata,
-                dagster_type=dagster_type,
-            )
-        return output
```

Comment:
pretty sure this code was just dead so refactored at [2] and clarified what is supported